### PR TITLE
ref(grouping): Remove fetch levels request & leadHint 

### PR DESCRIPTION
**Pull request by priscilawebdev**
. the "grouping" badge will only be displayed if the group current_level is provided and if it's not provided no extra request is needed to fetch it
. LeadHint is no longer necessary as the UI no longer displays in app - only by default but all relevant frames

### CODE REVIEW

File: static/app/components/eventOrGroupTitle.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ function EventOrGroupTitle({
   className,
 }: Props) {
   const event = data as Event;
-  const groupingCurrentLevel = (data as BaseGroup).metadata?.current_level;
+  const groupingCurrentLevel = (data as BaseGroup).metadata.current_level;
```
Comment (priscilawebdev) [inline static/app/components/eventOrGroupTitle.tsx:37]: metadata is always provided

File: static/app/components/events/interfaces/frame/lineV2/groupingBadges.tsx [hunk 2]
```diff
@@ -15,15 +15,15 @@ function GroupingBadges({isPrefix, isSentinel, isUsedForGrouping}: Props) {
   const badges: React.ReactElement[] = [];
 
   if (isSentinel) {
-    badges.push(<GroupingBadge badge={FrameBadge.SENTINEL} />);
+    badges.push(<GroupingBadge key={FrameBadge.SENTINEL} badge={FrameBadge.SENTINEL} />);
```
Comment (priscilawebdev) [inline static/app/components/events/interfaces/frame/lineV2/groupingBadges.tsx:18]: this was causing a react warning

File: static/app/components/events/interfaces/stacktraceContentV2.tsx [hunk 3]
```diff
@@ -239,15 +196,14 @@ function StackTraceContent({
...
-            nextFrame,
             prevFrame,
             platform,
             timesRepeated: nRepeats,
             showingAbsoluteAddress: showingAbsoluteAddresses,
             onAddressToggle: handleToggleAddresses,
             image: findImageForAddress(frame.instructionAddr, frame.addrMode),
             maxLengthOfRelativeAddress: maxLengthOfAllRelativeAddresses,
-            registers: {}, // TODO: Fix registers
+            registers: {},
```
Comment (priscilawebdev) [inline static/app/components/events/interfaces/stacktraceContentV2.tsx:206]: there is no need to fix registers here. I clarified this logic with Markus yesterday and he said that this is correct, as the registers should only passed to the last frame

File: static/app/components/events/interfaces/frame/lineV2/wrapper.tsx [hunk 4]
```diff
@@ -7,17 +7,29 @@ const Wrapper = styled('div')<{
...
       : p.haveFramesAtLeastOneExpandedFrame
-      ? '1fr 16px'
-      : '1fr'};
-  grid-gap: ${space(1)};
+      ? `
+          grid-template-columns: 1fr 16px;
+          grid-gap: ${space(1)};
+        `
+      : `
+          grid-template-columns: 1fr;
```
Comment (priscilawebdev) [inline static/app/components/events/interfaces/frame/lineV2/wrapper.tsx:27]: there was an extra spacing being shown on mobile when not needed

**Review by matejminar — APPROVED** [review]

File: static/app/components/eventOrGroupTitle.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/components/eventOrGroupTitle.tsx]: apparently not in our tests :/

**Pull request merged**